Share Screen crashes Zed

### Summary

<!-- Begin your issue with a one sentence summary -->

`Share Screen` seems to crash Zed for me. I've used it many times before with no issues. I'll add a video and the Apple crash report logs.

I'm not 100% sure but I think the Zed Log that references `SCShareableContent` is related to the problem.

### Description

https://github.com/user-attachments/assets/066752da-887e-4d76-9b22-8b5370659d82


Crash report logs link (too long for github comment):
https://gist.github.com/DavisVaughan/17a007ab6f875eaf41a8fbf2429efd27


### Zed Version and System Specs

Zed: v0.181.8 (Zed) 
OS: macOS 13.7.4
Memory: 32 GiB
Architecture: aarch64

Worth noting that I'm on a fairly new mac, but decently old version of macos (13x)

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Paste your log inside the code block. -->
```log
2025-04-15T11:15:42.705604-04:00 [INFO] ========== starting zed ==========
2025-04-15T11:15:42.741427-04:00 [INFO] Opening main db
2025-04-15T11:15:42.742349-04:00 [INFO] Opening main db
2025-04-15T11:15:42.765434-04:00 [INFO] Using git binary path: Some("/Applications/Zed.app/Contents/MacOS/git")
2025-04-15T11:15:42.809504-04:00 [INFO] No prompt template overrides directory found at /Users/davis/.config/zed/prompt_overrides. Using built-in prompts.
2025-04-15T11:15:42.810018-04:00 [INFO] extensions updated. loading 4, reloading 0, unloading 0
2025-04-15T11:15:42.837339-04:00 [INFO] Opening main db
2025-04-15T11:15:42.864304-04:00 [INFO] set status on client 0: Authenticating
2025-04-15T11:15:42.894088-04:00 [INFO] set status on client 148268: Connecting
2025-04-15T11:15:42.894278-04:00 [INFO] Opening main db
2025-04-15T11:15:42.908778-04:00 [INFO] Opening main db
2025-04-15T11:15:42.913399-04:00 [INFO] Opening main db
2025-04-15T11:15:43.045619-04:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2025-04-15T11:15:43.441067-04:00 [INFO] set environment variables from shell:/bin/zsh, path:/Users/davis/.local/bin:/Users/davis/Library/pnpm:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Apple/usr/bin:/Applications/quarto/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/davis/.cargo/bin
2025-04-15T11:15:49.186327-04:00 [INFO] add_connection;
2025-04-15T11:15:52.48817-04:00 [INFO] set status on client 148268: Connected { peer_id: PeerId { owner_id: 730, id: 14359 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2025-04-15T11:15:59.114794-04:00 [INFO] Opening main db
2025-04-15T11:16:08.760298-04:00 [INFO] connecting to wss://zed.livekit.cloud/rtc?sdk=rust&protocol=15&auto_subscribe=1&adaptive_stream=0&version=0.7.8&access_token=...
2025-04-15T11:16:11.222915-04:00 [ERROR] {
  "thread": "main",
  "payload": "Class with name SCShareableContent could not be found",
  "location_data": {
    "file": "crates/gpui/src/platform/mac/screen_capture.rs",
    "line": 194
  },
  "backtrace": [
    "backtrace::capture::Backtrace::create::h4b69b6157ecf36cf+16779240",
    "backtrace::capture::Backtrace::new::hd5dfddeb689ca4ae+16779048",
    "zed::reliability::init_panic_hook::{{closure}}::h3a4edbbf05997323+105779264",
    "std::panicking::rust_panic_with_hook::hc9e2ec2a12b10e12+84433660",
    "std::panicking::begin_panic_handler::{{closure}}::h3b6449e2eb83bf4c+84432448",
    "std::sys::backtrace::__rust_end_short_backtrace::h38d85099ddd107a3+84426632",
    "_rust_begin_unwind+84431660",
    "core::panicking::panic_fmt::hd4003258f6ab7b29+122213820",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::screen_capture_sources::h748d06915f8b675c+45608460",
    "call::call_impl::room::Room::share_screen::h0e9eb04e3a31378a+18377348",
    "title_bar::collab::<impl title_bar::TitleBar>::render_call_controls::{{closure}}::he23001b35366c6a6+89921844",
    "gpui::window::Window::on_mouse_event::{{closure}}::h3705d525901f36af+45897692",
    "gpui::window::Window::dispatch_event::hd92939862d150ff6+45974772",
    "gpui::window::Window::new::{{closure}}::h641eca527e7c11ab+45858948",
    "gpui::platform::mac::window::handle_view_event::h8973a61d5205db4d+45678828",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run::h6486203a590d5e02+45604008",
    "gpui::app::Application::run::hba8d6b6e2c35bbe5+102906284",
    "zed::main::h70a7f16c4f246f32+105807132",
    "std::sys::backtrace::__rust_begin_short_backtrace::h4e98eea6c8752f7b+105632488",
    "std::rt::lang_start::{{closure}}::hc80c2e9b5ae4562b+104846492",
    "std::rt::lang_start_internal::h5f91760815528aa2+84368772",
    "_main+105820412"
  ],
  "app_version": "0.181.8",
  "app_commit_sha": "c294000cfa619366cb836251862720248cf0eea3",
  "release_channel": "stable",
  "target": "aarch64-apple-darwin",
  "os_name": "macOS",
  "os_version": "13.7.4",
  "architecture": "aarch64",
  "panicked_on": 1744730171222,
  "system_id": "fc6a5ca9-3856-4e5c-b6b7-d589fad3ee90",
  "installation_id": "1f48f6a2-0ea5-4819-aef1-16e0eb05c12a",
  "session_id": "1b956bda-57fa-4cdd-9408-631b6f87e6fb"
}
2025-04-15T11:16:13.031449-04:00 [INFO] ========== starting zed ==========
```

</details>

--- a/crates/gpui/build.rs
+++ b/crates/gpui/build.rs
@@ -77,8 +77,6 @@ mod macos {
 
     fn generate_dispatch_bindings() {
         println!("cargo:rustc-link-lib=framework=System");
-        // weak link to support Catalina
-        println!("cargo:rustc-link-arg=-Wl,-weak_framework,ScreenCaptureKit");
 
         let bindings = bindgen::Builder::default()
             .header("src/platform/mac/dispatch.h")

--- a/crates/zed/build.rs
+++ b/crates/zed/build.rs
@@ -12,6 +12,9 @@ fn main() {
 
         // Register exported Objective-C selectors, protocols, etc
         println!("cargo:rustc-link-arg=-Wl,-ObjC");
+
+        // weak link to support Catalina
+        println!("cargo:rustc-link-arg=-Wl,-weak_framework,ScreenCaptureKit");
     }
 
     // Populate git sha environment variable if git is available

